pkg/providers/v1/instances_v2.go

```diff
@@ -74,9 +79,32 @@ func (c *Cloud) getAdditionalLabels(zoneName string) (map[string]string, error)
 
 	additionalLabels[LabelZoneID] = zoneID
 
+	if c.topologySupportedInstance(instanceType) && slices.Contains(c.cfg.Global.TopologySupportedRegions, region) {
+		topologyRequest := &ec2.DescribeInstanceTopologyInput{InstanceIds: []*string{&instanceID}}
+		topology, err := c.ec2.DescribeInstanceTopology(topologyRequest)
+		if err != nil || topology == nil {
+			klog.Infof("topology api not supported for instance type: %s or region: %s", instanceType, region)
+			return additionalLabels, nil
+		}
+		for index, networkNode := range topology.NetworkNodes {
+			label := LabelNetworkNode + strconv.Itoa(index)
+			additionalLabels[label] = *networkNode
+		}
+	}
+	
 	return additionalLabels, nil
 }
 
+// TopologySupportedInstancePrefixes is help to filter instance types supported by topology API
+func (c *Cloud) topologySupportedInstance(instanceType string) bool {
+	for _, prefix := range c.cfg.Global.TopologySupportedInstancePrefixes {
+		if strings.HasPrefix(instanceType, prefix) {
```

Comment:
Do we want to be updating the cloud-provider each time there's a new AWS instance type, though?

Reply:
I discussed this with Carter before. We don't want to trigger this api for all nodes, because some cx might have hundreds or thousands of nodes in their cluster, if we call this api for all nodes, it very likely trigger throttle issue. 

And DescribeInstanceTopology only supported for very limited instances types which lots of cx might not use. check the supported instance type and regions here. https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceTopology.html

Reply:
> Do we want to be updating the cloud-provider each time there's a new AWS instance type, though?

We don't need to update cloud-provider, TopologySupportedInstanceTypes and TopologySupportedRegions is added in cloud-config where will accept the input in KCP CCM manifest. We only need to update the manifest with the cloud-config input.